### DISCUSSION TITLE
How does loco server works on deployment?

### DISCUSSION THREAD

**Discussion by arferreira** [loco-rs/loco#161] [Q&A] [answered]
Hello everyone, I tried to deploy my application using loco, but the server doesn't starts on production or when I ran the docker container. 

How can I start this in production environment?

**Comment by arferreira** [comment]
The same thing happens when running through docker compose:

```
 ✔ Container veeda_api-veeda_api-1  Created                                                                              0.0s 
Attaching to veeda_api-1
veeda_api-1  | The one-person framework for Rust
veeda_api-1  | 
veeda_api-1  | Usage: veeda_api [OPTIONS] <COMMAND>
veeda_api-1  | 
veeda_api-1  | Commands:
veeda_api-1  |   start     Start an app
veeda_api-1  |   db        Perform DB operations
veeda_api-1  |   routes    Describe all application endpoints
veeda_api-1  |   task      Run a custom task
veeda_api-1  |   generate  code generation creates a set of files and code templates based on a predefined set of rules
veeda_api-1  |   help      Print this message or the help of the given subcommand(s)
veeda_api-1  | 
veeda_api-1  | Options:
veeda_api-1  |   -e, --environment <ENVIRONMENT>  Specify the environment [default: development]
veeda_api-1  |   -h, --help                       Print help
veeda_api-1  |   -V, --version                    Print version
veeda_api-1 exited with code 2
```

PS: I'm using the exact docker file created by the generate.

**Comment by kaplanelad** [comment]
Hey @arferreira,  it seems like you missed including the command for running the Docker container. However, you have two options to run your container in production mode:

Option 1: Utilize the -e Docker flag to pass environment variables.
```sh
docker run -e LOCO_ENV=production -p 3000:3000 [IMAGE-NAME] start
```

Option 2: Employ the --environment loco flag when initiating your Docker process.

```sh
docker run -p 3000:3000 [IMAGE-NAME] start --environment production
```

**Reply by arferreira** [reply]
Thank you so much @kaplanelad !

That's cool, but the problem now is that I created a database on AWS RDS and another one on Vercel. All of them I can't connect.

Error:

```
[+] Running 1/1                                                                                                                                         
 ✔ Container veeda_api-veeda_api-1  Recreated                                                                                                      0.1s 
Attaching to veeda_api-1
veeda_api-1  | Error: Connection Error: pool timed out while waiting for an open connection
veeda_api-1  | 
veeda_api-1  | Caused by:
veeda_api-1  |    0: pool timed out while waiting for an open connection
veeda_api-1  |    1: pool timed out while waiting for an open connection
veeda_api-1  | 
veeda_api-1  | Location:
veeda_api-1  |     /usr/local/cargo/registry/src/index.crates.io-6f17d22bba15001f/loco-rs-0.1.6/src/cli.rs:257:31
veeda_api-1 exited with code 1
```

A curious fact is that I can connect in both databases via command line or PgAdmin. But using loco I can't. Have you seen something like that?

**Reply by kaplanelad** [reply, marked as answer]
There may be a network issue between your server and your RDS. Please consider the following potential problems:

1. Verify that the connection string contains accurate details such as user, password, and database. (Please don't past it here.)
2. Check the server's and RDS's connectivity, such as security groups and network ACLs. If you have SSH access to the server, you can use tools like telnet or PostgreSQL CLI to check the connection.

If you have connectivity between the server and the RDS. in DB configuration under the `database` yaml section, you have `connect_timeout`, which gets the default of 500 milliseconds. try to increase to bigger number

**Reply by arferreira** [reply]
Hey @kaplanelad, thanks a lot. 

So the problem was on connect_timeout, increasing by 5000, I could run the migrations and the server. 
Should have some layers on AWS side to check the connection. I will test now deploying in my ECS to get my first version using Loco. I rewrote an entire backend which was wrote in PHP using Loco.

**Comment by kaplanelad** [comment]
@arferreira, No problem :) I'm happy to help. If you need more help, we are here to assist you. 

BTW, what kind of backend is it?

**Reply by arferreira** [reply]
@kaplanelad It is clinical management and telemedicine software.

**Reply by kaplanelad** [reply]
Thank you, @arferreira, for sharing!  We'd love to hear more about your experience with Loco, including:

- Loco's performance
- Your initial experience starting with Loco
- How was to start with Loco
- What was good?
- and most importantly what we should improve

We just opened a Discord, so feel free to ping me privately 
https://discord.gg/Npcwuvq6

**Reply by arferreira** [reply]
@kaplanelad I share my feedback on discord. I'm excited to contribute with you as soon as I can. 
Thanks again.
